### DISCUSSION TITLE
Expression syntax for not startswith

### DISCUSSION THREAD

**Discussion by phw** [community/community#25017] [Actions]
<p>I have some trouble figuring out the syntax of Github Action expressions, and the documentation is not very clear on this. I know I can run a step only for tagged commits with</p>
<pre><code class="lang-auto">
if: startsWith(github.ref, 'refs/tags/')

</code></pre>
<p>How do I inverse this, so it is only run if it is not a tag? The docs for expressions say <code>!</code> is the not operator, but something like <code>! startsWith(github.ref, 'refs/tags/')</code> or <code>!startsWith(github.ref, 'refs/tags/')</code> gives a syntax error.</p>

**Comment by jherico** [comment, marked as answer]
<p>have you tried </p>
<p><span>if: startsWith(github.ref, ‘refs/tags/’) != true</span></p>
<p>or</p>
<p><span>if: !(startsWith(github.ref, ‘refs/tags/’))</span></p>

**Comment by phw** [comment]
<p>Thanks a lot, this worked:</p>
<p><span>if: startsWith(github.ref, ‘refs/tags/’) != true</span></p>
<p>The other variant with !(…) I had already tried, it gives a syntax error.</p>

**Comment by Yanjingzhu** [comment]
<p>If you want to use ! in front of startsWith, you can put whole if condition inside double quotes:</p>
<pre><code>if: "!startsWith(github.ref, 'refs/tags/')"
</code></pre>

**Comment by jeking3** [comment]
<p>It would be useful for GitHub to add an example.  I had a hard time finding this.</p>

**Comment by mloskot** [comment]
Another one:

```yaml
if: ${{ ! startsWith(github.ref, 'refs/tags/') }}
```

see detailed explanation in [If (not) startswith mutually exclusive steps #26386](https://github.com/orgs/community/discussions/26386#discussioncomment-3251726).
